autoload/clap.vim

```diff
@@ -388,6 +388,15 @@ function! clap#(bang, ...) abort
   call clap#for(provider_id_or_alias)
 endfunction
 
+function! clap#calculate_preview_direction()
+  if g:clap_preview_direction ==# 'AUTO' 
+    let s:preview_direction = &columns < 80 ? 'UD' : 'LR'
+  else 
+    let s:preview_direction = g:clap_preview_direction
+  endif
+  return s:preview_direction
+endfunction
```

Comment:
This can be moved to autoload/clap/preview.vim.

```viml
function! clap#preview#direction()
  if g:clap_preview_direction ==# 'AUTO'
    return &columns < 80 ? 'UD' : 'LR'
  else
    return g:clap_preview_direction
  endif
endfunction
```